Python/sysmodule.c

```diff
@@ -3358,6 +3364,19 @@ make_impl_info(PyObject *version_info)
     if (res < 0)
         goto error;
 
+#if !defined(MS_WINDOWS) && defined(HAVE_SYS_UTSNAME_H)
+    res = uname(&u);
```

Comment:
I would prefer to not add a syscall at every Python startup. You can already call `os.uname().machine` to get the information.

Reply:
I'm sorry, I posted a thread in discuss about this change, I have a commit that works on all platforms, but it's more suitable as a public C API.

https://discuss.python.org/t/about-pydoc-var-should-be-changed-to-be-invisible/64480/6

Reply:
What's the relationship with PyDoc_VAR?

Reply:
> What's the relationship with PyDoc_VAR?

Sorry, I seem to have given the wrong link, I have updated it now.

Reply:
As I wrote in your thread, I don't think that there is need for Py_ArchName().

This issue is about the Python API sys.implementation. It's confusing that you are now speaking about a C API. It would be better if this PR is focused on the Python API.

Reply:
> As I wrote in your thread, I don't think that there is need for Py_ArchName().
> 
> This issue is about the Python API sys.implementation. It's confusing that you are now speaking about a C API. It would be better if this PR is focused on the Python API.

Thanks for your guidance, I will refine this PR to make it compatible with all systems (By the way, this change is to make the implementation description more comprehensive, because the attributes in the implementation have corresponding python api).

Reply:
> Thanks for your guidance, I will refine this PR to make it compatible with all systems

I would prefer to not call uname() at Python startup. So I would prefer to make the change specific to Windows.

Reply:
> > Thanks for your guidance, I will refine this PR to make it compatible with all systems
> 
> I would prefer to not call uname() at Python startup. So I would prefer to make the change specific to Windows.

Hmmm, It seems like this would be better since there is already an API for Python. So my next changes will make it work only on Windows (since Windows doesn't have os.uname).

Reply:
I don't think that we should add information retrieved at runtime in sys. Usually, sys is more information computed during the build.

Reply:
Perhaps as zooba mentioned, this would make the description of sys.implementation more complete(Maybe this doesn't mean much).

Reply:
> Yes, `sys.implementation` should only contain information computed during the build, such as the architecture that the build is intended to target. We shouldn't need runtime information about the platform - if the build is running at all, we are compatible enough to assume that we're on the architecture targeted at build time.

That is to say, you want the value of'architecture to be consistent with sys. winver, not the running architecture.

Reply:
```
>>> import sys;sys.implementation
namespace(name='cpython', cache_tag='cpython-314', version=sys.version_info(major=3, minor=14, micro=0, releaselevel='alpha', serial=0), hexversion=51249312, _architecture='3.14')
```

Reply:
"Consistent with" doesn't mean "the same as".

I really want it to be the same as the architecture in the installer filename - either `win32`, `amd64` or `arm64`, which should be the `$(ArchName)` property during build.

Reply:
That is, you want the value to belong to the following show
```
<ArchName Condition="'$(ArchName)' == '' and $(Platform) == 'x64'">amd64</ArchName>
<ArchName Condition="'$(ArchName)' == '' and $(Platform) == 'ARM'">arm32</ArchName>
<ArchName Condition="'$(ArchName)' == '' and $(Platform) == 'ARM64'">arm64</ArchName>
<ArchName Condition="'$(ArchName)' == ''">win32</ArchName>
```
I think I know what to do.

Reply:
You would use a preprocessor variable to pass in whatever value `$(ArchName)` has, as I referenced earlier for you. That way if we change it or add more later, it'll automatically show up in the build. There's no need to define an exhaustive list, and very little logic required other than setting it (which you obviously know how to do, it's just about where the value comes from).

Reply:
@zooba, look this
```
>>> sys.implementation
namespace(name='cpython', cache_tag='cpython-314', version=sys.version_info(major=3, minor=14, micro=0, releaselevel='alpha', serial=0), hexversion=51249312, _architecture='win32')
>>> sys.platform
'win32'
```
Is this what you wanted?

Reply:
For a 32-bit build, yeah.

Reply:
It seems that there is a Py_GetPlatform() in capi for getting the current platform, and it seems that its value matches ArchName (unfortunately I don't have armPC on hand).

Reply:
Afraid not, that gets `sys.platform`, which is "win32" for _all_ Windows builds. It never changes, because it means the Win32 API.

You will need to pass in the `$(ArchName)` variable via a new preprocessor variable to make it available when compiling.

Reply:
> Afraid not, that gets `sys.platform`, which is "win32" for _all_ Windows builds. It never changes, because it means the Win32 API.
> 
> You will need to pass in the `$(ArchName)` variable via a new preprocessor variable to make it available when compiling.

Okay, I'm going to implement the preprocessor directive scheme I wrote.

Reply:
> Yes, `sys.implementation` should only contain information computed during the build

Just to be clear, `sys.implementation` is information about the Python implementation, not the build.